// @huggingface/inference/dist/commonjs/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROVIDERS_HUB_ORGS = exports.PROVIDERS_OR_POLICIES = exports.INFERENCE_PROVIDERS = void 0;
exports.INFERENCE_PROVIDERS = [
    "baseten",
    "cerebras",
    "cohere",
    "deepinfra",
    "fal-ai",
    "featherless-ai",
    "fireworks-ai",
    "groq",
    "hf-inference",
    "novita",
    "nscale",
    "openai",
    "ovhcloud",
    "publicai",
    "replicate",
    "scaleway",
    "together",
    "wavespeed",
    "zai-org",
];
exports.PROVIDERS_OR_POLICIES = [...exports.INFERENCE_PROVIDERS, "auto"];
/**
 * The org namespace on the HF Hub i.e. hf.co/…
 *
 * Whenever possible, InferenceProvider should == org namespace
 */
exports.PROVIDERS_HUB_ORGS = {
    baseten: "baseten",
    cerebras: "cerebras",
    cohere: "CohereLabs",
    deepinfra: "DeepInfra",
    "fal-ai": "fal",
    "featherless-ai": "featherless-ai",
    "fireworks-ai": "fireworks-ai",
    groq: "groq",
    "hf-inference": "hf-inference",
    novita: "novita",
    nscale: "nscale",
    openai: "openai",
    ovhcloud: "ovhcloud",
    publicai: "publicai",
    replicate: "replicate",
    scaleway: "scaleway",
    together: "togethercomputer",
    wavespeed: "wavespeed",
    "zai-org": "zai-org",
};
